tfhe/src/integer/server_key/comparator.rs

```diff
@@ -188,6 +178,75 @@ impl<'a> Comparator<'a> {
         self.server_key.key.unchecked_scalar_add_assign(lhs, 1);
     }
 
+    // lhs will be assigned
+    // - 0 if lhs < rhs
+    // - 1 if lhs == rhs
+    // - 2 if lhs > rhs
+    fn scalar_compare_block_assign<PBSOrder: PBSOrderMarker>(
+        &self,
+        lhs: &mut crate::shortint::CiphertextBase<PBSOrder>,
+        scalar: u8,
+    ) {
+        // Same logic as compare_block_assign
+        // but rhs is a scalar
+        let delta =
+            (1u64 << (u64::BITS as u64 - 1)) / (lhs.carry_modulus.0 * lhs.message_modulus.0) as u64;
+        let plaintext = Plaintext((scalar as u64) * delta);
+        crate::core_crypto::algorithms::lwe_ciphertext_plaintext_sub_assign(&mut lhs.ct, plaintext);
+        self.server_key
+            .key
+            .apply_lookup_table_assign(lhs, &self.sign_accumulator);
+
+        self.server_key.key.unchecked_scalar_add_assign(lhs, 1);
+    }
+
+    fn reduce_two_sign_blocks_assign<PBSOrder: PBSOrderMarker>(
+        &self,
+        msb_sign: &mut crate::shortint::CiphertextBase<PBSOrder>,
+        lsb_sign: &crate::shortint::CiphertextBase<PBSOrder>,
+    ) {
+        // We don't use pack_block_assign as the offset '4' does not depend on params
+        self.server_key.key.unchecked_scalar_mul_assign(msb_sign, 4);
+        self.server_key.key.unchecked_add_assign(msb_sign, lsb_sign);
+
+        self.server_key
+            .key
+            .apply_lookup_table_assign(msb_sign, &self.selection_accumulator);
```

Comment:
It would, but we would do that later as the addition and non scalart_shift/rotate for example also need the algorithm to be specially rewrittenfor the 1_1 case

Reply:
can we re-purpose https://github.com/zama-ai/tfhe-rs-internal/issues/138 as an umbrella issue for all weird/partially supported param sets and create sub issues that are referenced in there ?

Reply:
Yes

Reply:
thanks !